File: ruby/json_de.rb
require "json"

json = '{
  "foo1": "bar",
  "foo2": 2,
  "foo3": [1, 2, 3],
  "foo4": null
}';
repeat_count = 1000000

start = Time.now
repeat_count.times { JSON.parse(json) }
_end = Time.now

puts ((_end - start) * 1000).to_i

File: javascript/json_de.js
const json = `{
  "foo1": "bar",
  "foo2": 2,
  "foo3": [1, 2, 3],
  "foo4": null
}`;
const repeatCount = 1000000;

const start = Date.now();
for (let i = 0; i < repeatCount; ++i) {
  JSON.parse(json);
}
const end = Date.now();

console.log(end - start);
